ntbs-service/Services/NotificationChangesService.cs

```diff
@@ -104,13 +104,19 @@ private IEnumerable<AuditLog> GetCanonicalLogs(List<AuditLog> group)
                 return group;
             }
 
-            // Social risks page contains sub-models, which create their own audit entries
+            // Social risks page contains sub-models if they are being set for the first time, which create their own audit entries
             var socialRisksUpdate = group.Find(log => log.EntityType == nameof(SocialRiskFactors));
             if (socialRisksUpdate != null)
             {
                 return new List<AuditLog> {socialRisksUpdate};
             }
 
+            // Social risks page updates can create a group of only risk factor detail logs
+            if (group.All(log => log.EntityType == nameof(RiskFactorDetails)))
+            {
+                return new List<AuditLog> {group.First()};
+            }
+
```

Comment:
Perhaps it would be safer to just say that any group containing either `SocialRiskFactors` or `RiskFactorDetails` is allowed? I'd be happier with a single condition here, because it's really the same concern.

Reply:
Made this change, there is just one condition for risk factors now

Reply:
Great, looks good